extensions/banner/deployment/src/main/java/io/quarkus/banner/deployment/BannerConfig.java

```diff
@@ -0,0 +1,14 @@
+package io.quarkus.banner.deployment;
+
+import io.quarkus.runtime.annotations.ConfigItem;
+import io.quarkus.runtime.annotations.ConfigPhase;
+import io.quarkus.runtime.annotations.ConfigRoot;
+
+@ConfigRoot(name = "banner", phase = ConfigPhase.BUILD_TIME)
+public class BannerConfig {
+    /**
+     * The path of the banner which could be provided by user
+     */
+    @ConfigItem(defaultValue = "default_banner.txt")
+    public String path;
```

Comment:
I think there should be different paths depending on the console color support (no color, 8 color, 16 color, 256 color, truecolor) that are searched in order from most to least color support (since color modes are backwards compatible).

One thing I had looked into which might be interesting to explore further was rendering PNG images to full color text banners.  I have a working(ish) prototype around somewhere for 256 and 16M colors.  But that's a separate discussion I guess!

Reply:
That would be completely new. IG. Could you give me little leads for this ? 😅

Reply:
For the color stuff... it looks like we don't (yet) expose that information, so it might be too hard to utilize for now.  But, we can make a note for later. :slightly_smiling_face: